pkg/apis/garden/v1beta1/helper/helpers.go

```diff
@@ -163,6 +163,8 @@ func GetMachineImageNameFromShoot(cloudProvider gardenv1beta1.CloudProvider, sho
 		return shoot.Spec.Cloud.Alicloud.MachineImage.Name
 	case gardenv1beta1.CloudProviderOpenStack:
 		return shoot.Spec.Cloud.OpenStack.MachineImage.Name
+	case gardenv1beta1.CloudProviderLocal:
```

Comment:
Do we really need this part?

Reply:
I think so, otherwise `Botanist.RequiredExtensionsExist` ([link](https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/botanist.go#L165)) fails, because it does not find a matching `OperatingSystemConfig` resource, since the one in the function's `requiredExtensions` map variable is for machine name `""`.

Reply:
OK, agreed. Thank you!